internal/controller/deployment_controller.go

```diff
@@ -224,6 +285,20 @@ func (r *DeploymentReconciler) Update(ctx context.Context, l logr.Logger, deploy
 				Message: hrReadyCondition.Message,
 			})
 		}
+
+		requeue, err := r.setStatusFromClusterStatus(ctx, dc, l, deployment)
+		if err != nil {
+			if requeue {
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+			} else {
+				return ctrl.Result{}, err
+			}
+		}
+
+		if requeue {
+			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Instead of requeuing every 10 seconds, perhaps its better to add a `Watches` for the cluster object (once it's added to scheme) within `SetupWithManager`, that way when the status of the cluster object changes we would requeue rather then relying on polling.

Reply:
Looks like there's only one watch allowed per controller unfortunately.